.github/workflows/ci.yml

```diff
@@ -29,12 +34,49 @@ jobs:
           sudo rm -rf "$AGENT_TOOLSDIRECTORY"
           echo "=== After cleanup:"
           df -h
-      - uses: actions/checkout@v3
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        # see https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ${{ env.REGISTRY }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+        # The .ts scripts depend upon this
       - run: yarn global add tsx
+
       - name: build test images
-        run: ./buildTestImages.ts
+        run: |
+          docker info
+          ./buildTestImages.ts
       - name: run test images
         run: ./runTestImages.ts
+
+      # XXX this should be instant because all the stages were already built in the steps above
+      # but it's re-building the last stage. This is deemed good enough for now.
+      # see https://github.com/moby/moby/issues/34715
+      - name: Build and push complete image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          # push to registry on every repo push. A PR #2 will push with tag `pr-2` and `main` will have tag `main`.
```

Comment:
When should I expect `:latest` to get updated? Is `:latest` a thing at all in this workflow?

In [ci logs](https://github.com/Agoric/agoric-3-proposals/actions/runs/6772404372/job/18404886442?pr=11), I see

```
#73 pushing manifest for ghcr.io/agoric/agoric-3-proposals:pr-11@sha256:0f8745ce487cf505d545e5fcc98266583928baba42dac7336e43ade30c801144 2.0s done
```

I tried `docker pull ghcr.io/agoric/agoric-3-proposals:latest`... I got something, but it doesn't match:
```
latest: Pulling from agoric/agoric-3-proposals
Digest: sha256:af7b4c7936c836e63e03ea797923513d81d48279de09397659976e4927f1e27c
Status: Image is up to date for ghcr.io/agoric/agoric-3-proposals:latest
ghcr.io/agoric/agoric-3-proposals:latest
```

I then tried `docker pull ghcr.io/agoric/agoric-3-proposals:pr-11` and it matched, after downloading a bunch of stuff:

<details><summary>pulling agoric-3-proposals:pr-11</summary>

```
pr-11: Pulling from agoric/agoric-3-proposals
6a70103cc499: Already exists 
ccb4221e2c63: Already exists 
9fe6e67e69c8: Already exists 
ea30286b46fb: Already exists 
f75d822ab76b: Already exists 
2a87910e6b0c: Already exists 
ab305ba4cb89: Already exists 
88cbdb8927c8: Already exists 
834c64460156: Already exists 
4f4fb700ef54: Pulling fs layer 
7e8324770d8b: Already exists 
e192c61a6def: Already exists 
c1d4948858f7: Already exists 
5a814c9274ba: Already exists 
346b2a71a005: Already exists 
ab4ff57efef0: Already exists 
a03ddbfe4d35: Already exists 
10adcfe6c382: Already exists 
4f4fb700ef54: Pull complete 
d09e22f9ea6a: Already exists 
12b856665564: Already exists 
e9b5f3e6f97d: Already exists 
facabfcce2e2: Pull complete 
e4439afb5d15: Pull complete 
56d24d137f12: Pull complete 
8fe61e7e41f7: Pull complete 
7f80f1216b96: Pull complete 
ae78e7f4d96c: Pull complete 
98c37641cd3b: Pull complete 
2f4b49cbabcf: Pull complete 
455ff1e7f43e: Pull complete 
a0cd10d7f86b: Pull complete 
381fe0cf8d26: Pull complete 
fe815550ea19: Pull complete 
Digest: sha256:0f8745ce487cf505d545e5fcc98266583928baba42dac7336e43ade30c801144
Status: Downloaded newer image for ghcr.io/agoric/agoric-3-proposals:pr-11
ghcr.io/agoric/agoric-3-proposals:pr-11
```
</details>

Reply:
`latest` was a thing in an earlier revision of this PR. I took it out because the term is generally used for "latest release" and there's no release. The defaults of this step are to use the default branch name (`main`) which I think is clearer. I'll document in README.